pyccl/tracers.py

```diff
@@ -264,6 +265,157 @@ def get_bessel_derivative(self):
+                mg_transfer_new = (mg_transfer[0], mg_transfer[1] *
+                                   bias_transfer_a[1])
+                self.add_tracer(cosmo, kernel, transfer_a=mg_transfer_new,
+                                der_bessel=der_bessel, der_angles=der_angles,
+                                transfer_k=bias_transfer_k)
+            else:
+                # Scale-independent case where k=None
+                mg_transfer_new = (mg_transfer[0], mg_transfer[1] *
+                                   bias_transfer_a[1])
+                self.add_tracer(cosmo, kernel, transfer_a=mg_transfer_new,
+                                der_bessel=der_bessel, der_angles=der_angles,
+                                transfer_k=bias_transfer_k)
+        #  case of an astro bias depending on a but not k
+        elif ((bias_transfer_a is not None) and (bias_transfer_k is None)):
+            if (isinstance(k, (list, np.ndarray))):
+                # Scale-dependent case with full k array
+                mg_transfer_new = (mg_transfer[0], mg_transfer[1],
+                                   (bias_transfer_a[1] * mg_transfer[2].T).T)
+                self.add_tracer(cosmo, kernel, transfer_ka=mg_transfer_new,
+                                der_bessel=der_bessel, der_angles=der_angles)
+            elif (isinstance(k, float) or isinstance(k, int)):
+                # Scale-dependent case with  a single specified k value
+                mg_transfer_new = (mg_transfer[0], mg_transfer[1] *
+                                   bias_transfer_a[1])
+                self.add_tracer(cosmo, kernel, transfer_a=mg_transfer_new,
+                                der_bessel=der_bessel, der_angles=der_angles)
+            else:
+                # Scale-independent case where k=None
+                mg_transfer_new = (mg_transfer[0], mg_transfer[1] *
+                                   bias_transfer_a[1])
+                self.add_tracer(cosmo, kernel, transfer_a=mg_transfer_new,
+                                der_bessel=der_bessel, der_angles=der_angles)
+        #  case of an astro bias depending on k but not a
+        elif ((bias_transfer_a is None) and (bias_transfer_k is not None)):
+            if (isinstance(k, (list, np.ndarray))):
+                # Scale-dependent case with full k array
+                mg_transfer_new = (mg_transfer[0], mg_transfer[1],
+                                   (bias_transfer_k[1] * mg_transfer[2]))
+                self.add_tracer(cosmo, kernel, transfer_ka=mg_transfer_new,
+                                der_bessel=der_bessel, der_angles=der_angles)
+            elif (isinstance(k, float) or isinstance(k, int)):
+                # Scale-dependent case with  a single specified k value
+                self.add_tracer(cosmo, kernel, transfer_a=mg_transfer,
+                                der_bessel=der_bessel, der_angles=der_angles,
+                                transfer_k=bias_transfer_k)
+            else:
```

Comment:
As before, I think this can be merged with the previous one

Reply:
condensed